apps/core/lib/core/pubsub/protocols/fanout.ex

```diff
@@ -45,6 +45,18 @@ defimpl Core.PubSub.Fanout, for: Core.PubSub.UserCreated do
   end
 end
 
+defimpl Core.PubSub.Fanout, for: Core.PubSub.UserUpdated do
+  alias Core.Services.Users
+
+  def fanout(%{item: %{email_changed: true} = user}) do
+    confirm_by = Timex.now() |> Timex.shift(days: 7)
+    Users.update_user(%{email_confirmed: false, email_confirm_by: confirm_by}, user)
```

Comment:
this logic should be added to the changeset in `Core.Schema.User` actually

Reply:
It's done.